docs/Examples2/QRCode.example.purs

```diff
@@ -1,53 +1,75 @@
 import Effect.Unsafe (unsafePerformEffect)
 import Lumi.Components (lumiElement)
 import Lumi.Components.Example (example)
+import Lumi.Components.Input as Input
 import Lumi.Components.Spacing (Space(..), vspace)
-import Lumi.Components.Text (p_)
+import Lumi.Components.Text (subsectionHeader_)
 import Lumi.Components2.Box (box)
-import Lumi.Components2.Button (button)
-import Lumi.Components2.Button as Button
+import Lumi.Components2.Link (link)
 import Lumi.Components2.QRCode (ErrorCorrectLevel(..), useQRCode)
 import Lumi.Styles as S
 import Lumi.Styles.Border as Border
-import React.Basic.DOM as R
-import React.Basic.Hooks (JSX, ReactComponent, component, element)
+import Lumi.Styles.Box (FlexAlign(..))
+import Lumi.Styles.Box as Box
+import React.Basic.DOM.Events (capture, targetValue)
+import React.Basic.Hooks (JSX, ReactComponent, component, element, useState, (/\))
 import React.Basic.Hooks as React
+import Web.HTML.History (URL(..))
 
 docs :: JSX
 docs =
-  lumiElement box
-    _
-      { content =
-        [ p_ "A QR Code pointing to \"https://www.lumi.com\""
-        , vspace S24
-        , example
-            $ element qrcodeExample { value: "https://www.lumi.com" }
-        ]
-      }
+  flip element {}
+    $ unsafePerformEffect do
+        component "QRCodeExample" \_ -> React.do
+          value /\ setValue <- useState "https://www.lumi.com"
+          pure
+            $ lumiElement box
+                _
+                  { content =
+                    [ Input.input
+                        Input.text_
+                          { value = value
+                          , onChange = capture targetValue $ traverse_ (const >>> setValue)
+                          }
+                    , vspace S24
+                    , example
+                        $ element qrcodeExample { value }
+                    ]
+                  }
 
 qrcodeExample :: ReactComponent { value :: String }
 qrcodeExample =
   unsafePerformEffect do
     component "QRCode" \props -> React.do
-      { qrcode, saveOnClick } <- useQRCode props.value L
+      { qrcode, url } <- useQRCode ECLLow props.value
       pure
         $ lumiElement box
-            _
-              { content =
-                [ lumiElement box
-                    <<< Border.border
-                    >>> Border._round
-                    >>> S.styleModifier_ (S.css { padding: S.int 16 })
-                    $ _ { content = [ qrcode ] }
-                , vspace S8
-                , lumiElement button
-                    <<< Button._secondary
-                    $ _
-                        { content = [ R.text "Download SVG" ]
-                        , onPress = saveOnClick "qrcode.svg"
-                        }
-                ]
-              }
+        <<< Box._align Center
+        $ _
+            { content =
+              [ lumiElement qrcode
+                  <<< Border.border
+                  >>> Border._round
+                  >>> S.styleModifier_
```

Comment:
Is there any reason why we need `<<<` and `>>>` here? I thought these modifiers worked fine just chained with `$`.

Reply:
I personally prefer `$` as I find it easier to read.

Reply:
Yeah I think having both `>>>` and `<<<` together in the same expression is probably best avoided.

Reply:
Hmm.. the direction matters though. Using all `>>>` or `$` causes the default border styles to get merged over this custom padding. I don't like that it works this way though, if you have any alternative ideas.

Reply:
The `<<<` composes a single style modifier into the component, while the `>>>` composes those 3 style modifiers into that single modifier. I think it'd be confusing to compose the styles upside down.

Reply:
Actually, I just tried flipping the definition of `propsModifier` to `(<<<)` and it now composes correctly with `$`. Maybe we want to make that change everywhere? It might break things. It should probably be a separate PR so testing that isn't holding other projects up.

Reply:
Oh I thought it has always been `(<<<)`? I don't think it'd be confusing, since `$` always reads right-to-left and has a pretty clear meaning IMO from function application. But I'm probably way too biased in this subject 😅

Reply:
Yeah, I guess I always imagined it like object merging, and with `$` and `<<<` it felt like the starting point was the modifier. I agree though, `$` everywhere is better.

Reply:
https://github.com/lumihq/purescript-lumi-components/issues/135